src/main/java/com/aws/greengrass/deployment/ShadowDeploymentListener.java

```diff
@@ -448,7 +448,7 @@ protected void shadowUpdated(Map<String, Object> desired, Map<String, Object> re
         if (cancelDeployment) {
             deployment = new Deployment(DeploymentType.SHADOW, UUID.randomUUID().toString(), true);
         } else {
-            deployment = new Deployment(fleetConfigStr, DeploymentType.SHADOW, configurationArn);
+            deployment = new Deployment(fleetConfigStr, DeploymentType.SHADOW, configuration.getDeploymentId());
```

Comment:
Should this be config arn instead of deployment id?

Reply:
No, this is supposed to be deployment id and jobs deployments set it accordingly. Shadow deployments set the config arn even though deployment id is available, may have been a mistake but this is why there are inconsistencies between jobs and shadow deployments